src/discussions/posts/data/selectors.js

```diff
@@ -1,20 +1,36 @@
 /* eslint-disable import/prefer-default-export */
 
-export const selectTopicThreads = topicId => state => (state.threads.topicThreadMap[topicId] || []).map(
-  threadId => state.threads.threads[threadId],
+export const selectTopicThreads = topicId => state => (state.threads.threadsInTopic[topicId] || []).map(
+  threadId => state.threads.threadsById[threadId],
 );
 
-export const selectThread = threadId => state => (state.threads.threads?.[threadId]);
+export const selectThread = threadId => state => (state.threads.threadsById?.[threadId]);
```

Comment:
You are not using `useModel` / `addModel` pattern here - whichi I guess most of the MFE use including course-authoring. Can you explain why?

Reply:
The hierarchy is more complex here, and isn't repeated across multiple models. So the existing useModel framework doesn't work well here. 

Currently, useModel will just save objects by id, however it doesn't care about other id references, for example, threadId or parentId.  Here we need to look up comments by their id, but also by their thread id or parent id. 

I've used [normalizr](https://github.com/paularmstrong/normalizr) for things like this before, and that might be another way to go.

Reply:
We looked at this normalizr implementation together and I think we can go with this and see how clean code gets after using it.